File: Algorithms/elucid.py
# Elucid algorithm that finds the greatest common divisor of two numbers
def elucid(number1, number2):
    # Identify the larger and the smaller of the two numbers
    larger = max(number1, number2)
    smaller = min(number1, number2)

    # Divide the larger number by the smaller number to determine if the smaller number
    # is the greatest common divisor
    remainder = larger % smaller
    if remainder == 0:
        return smaller
    else:
        return elucid(smaller, remainder) # Recursively call the Elucid algorithm


# Test the Elucid algorithm
print(elucid(5, 15))
print(elucid(10532, 1535))
print(elucid(24, 16))
print(elucid(24, 6))

File: Algorithms/elucid.cpp
#include <iostream>
#include <string>
#include <algorithm>

// Function declaration
int elucid(float x, float y);

int main() {
    int a = elucid(100, 20);
    std::cout << a << std::endl;
    system("pause");
}

// Function definition
// Given two numbers, return the great common divisor of the two numbers
int elucid(float x, float y) {
    int large = std::max(x, y);
    int small = std::min(x, y);   
    int remainder = large % small;
    if (remainder == 0) {
        return small;
    }
    return elucid(remainder, small);
}